tests/cbmc/sources/make_common_datastructures.c

```diff
@@ -159,48 +153,51 @@ EVP_MD_CTX* cbmc_allocate_EVP_MD_CTX() {
     return ctx;
 }
 
-struct s2n_hash_state* cbmc_allocate_s2n_hash_state()
-{
-    struct s2n_hash_state *state = malloc(sizeof(*state));
-    if (state != NULL)
-    {
-        state->hash_impl = malloc(sizeof(*(state->hash_impl)));
-        state->digest.high_level.evp.md = malloc(sizeof(*(state->digest.high_level.evp.md)));
-        state->digest.high_level.evp.ctx = cbmc_allocate_EVP_MD_CTX();
-        state->digest.high_level.evp_md5_secondary.md = malloc(sizeof(*(state->digest.high_level.evp_md5_secondary.md)));
-        state->digest.high_level.evp_md5_secondary.ctx = cbmc_allocate_EVP_MD_CTX();
+void cbmc_populate_s2n_evp_digest(struct s2n_evp_digest *evp_digest) {
+    if (evp_digest != NULL) {
+        /* `evp_digest->md` is never allocated.
+           It is always initialized based on the hashing algorithm.
+           If required, this initialization should be done in the validation function.
+        */
+        evp_digest->ctx = cbmc_allocate_EVP_MD_CTX();
     }
-    return state;
 }
 
 struct s2n_evp_digest* cbmc_allocate_s2n_evp_digest()
 {
     struct s2n_evp_digest *evp_digest = malloc(sizeof(*evp_digest));
-    if (evp_digest != NULL)
-    {
-        evp_digest->md = malloc(sizeof(*(evp_digest->md)));
-        evp_digest->ctx = malloc(sizeof(*(evp_digest->ctx)));
-    }
+    cbmc_populate_s2n_evp_digest(evp_digest);
     return evp_digest;
 }
 
+void cbmc_populate_s2n_hash_state(struct s2n_hash_state* state)
+{
+    if (state != NULL) {
+        /* `state->hash_impl` is never allocated.
+           It is always initialized based on the hashing algorithm.
+           If required, this initialization should be done in the validation function.
+        */
```

Comment:
I initially only commented on one of your comments, but I'd prefer if they all match the other comments in S2N. Can you audit your code for comment style?

(The nice thing about the style where every line starts with * is that it makes it very clear which lines are comments and which lines are code, especially for large comments)

Reply:
Ah yes, sorry, I should have checked. Will make this change and audit for comment style. Thanks!